File: java/ArmstrongNumbers.java
public class ArmstrongNumbers {
    public static void main(String[] args) {
        for (int i = 1; i < 1000; i++) {
            if (isArmstrong(i)) {
                System.out.print(i + " ");
            }
        }
    }

    public static boolean isArmstrong(int num) {
        int sum = 0, temp = num, n = 0;
        while (temp != 0) {
            temp /= 10;
            n++;
        }
        temp = num;
        while (temp != 0) {
            int digit = temp % 10;
            sum += Math.pow(digit, n);
            temp /= 10;
        }
        return num == sum;
    }
}

File: JavaScript/armstrong_numbers.js
function isArmstrong(num) {
    let digits = num.toString().split('');
    let n = digits.length;
    return num === digits.reduce((acc, digit) => acc + Math.pow(parseInt(digit), n), 0);
}

let armstrongNumbers = [];
for (let i = 1; i < 1000; i++) {
    if (isArmstrong(i)) armstrongNumbers.push(i);
}

console.log(armstrongNumbers);
